nmrfx-core/src/main/java/org/nmrfx/datasets/DatasetRegion.java

```diff
@@ -296,8 +300,18 @@ public int compareTo(Object o2) {
         return compare(this, o2);
     }
 
+    @Override
+    public int hashCode() {
+        int hash = 7;
+        hash = 31 * hash + (x == null ? 0 : Double.hashCode(x[0]));
+        return hash;
```

Comment:
Changed to `return x == null ? Objects.hashCode(x) : Double.hashCode(x[0]);` since just using Objects.hashCode(x) alone failed the unittest checking if equal objects have the same hashcode, since the comparison is made on only the first element of x, not sure if this is okay?

Reply:
I missed the trap in the equals. Your change looks correct.
For info, Objects.hashCode(null) returns 0.